### PR TITLE
build: add test and build scripts

### PR DESCRIPTION
**Pull request by sergio-costas**
This MR adds two little scripts that I use to build locally the snap, and to compile and launch the daemon to do quick tests. They have nothing "new", just saves the time of doing the steps specified in the HACKING.md files manually.

### CODE REVIEW

**Comment by sergio-costas** [comment]
I've been using these scripts for a while, and I think that maybe they could be useful for other people...

File: launch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+build() {
+    mkdir -p build
```
Comment (ernestl) [inline launch:4]: Do we specifically not want to build in /tmp?

**Comment by ernestl** [comment]
> I've been using these scripts for a while, and I think that maybe they could be useful for other people...

Looks like a nice time saver

**Review by ernestl — APPROVED** [review]

File: launch (hunk 1 above)
Comment (sergio-costas) [reply launch:4]: Since this is launched from the repository folder, it won't be built at /tmp...

Comment (sergio-costas) [reply launch:4]: (unless what you mean is that we want to build it at /tmp...)

**Comment by Meulengracht** [comment]
I'm not so sure about wanting these kind of helper scripts lying around in the root folder, second of all, since these scripts are rather simple (and more or less are automation of shell commands), is there a reason this couldn't just be a section in the `HACKING` document with a tutorial or a guide on how to do this, so people could create their own scripts if they need/want such a thing?

**Comment by sergio-costas** [comment]
It's another possibility, of course. Or moving them into a folder... The main point is that I used for a long time the remote build because I didn't know how to build it locally, and it was only when I got tired of the excessive time that it needed to build the snaps that I decided to investigate how to do it locally using multipass. It's not completely obvious because, for example, xenial images aren't listed in multipass. Also, moving from deb lxd to snap lxd has some little tricks involved...

**Comment by sergio-costas** [comment]
But, of course, an alternative is to just "copy" these scripts inside the HACKING.md file.

**Comment by sergio-costas** [comment]
or move them inside a folder (like "helper-scripts")

**Comment by Meulengracht** [comment]
The primary issue I have with these kinds of scripts is that usually no-one really have responsibility for them, so if anything changes for the script then the script breaks and suddenly we just have useless scripts lying around if no-one updates it.

It makes more sense to have these things documented (i.e how to build, or how to use your own build of snapd daemon for debugging) in the `HACKING` document, so if anyone needs to automate these kinds of things, the needed steps to do so are already documented and you can easily create a script.

**Review by Meulengracht — CHANGES_REQUESTED** [review]
Instead of scripts, we want to make sure the needed documentation for creating such scripts exists in `HACKING.md`

**Comment by sergio-costas** [comment]
That's a good point, indeed.

File: launch (hunk 1 above)
Comment (ernestl) [reply launch:4]: I am asking if we should do it the same as currently written in HACKING.md [here](https://github.com/snapcore/snapd/pull/12620/files#diff-9769a3f5504af3cd735e08f078b94f827bdd51eacac2e9d80ac9761bc485171fR206)

Comment (sergio-costas) [reply launch:4]: Mmm... good point

**Comment by ernestl** [comment]
We were reminded in stand-up recently to be consistant with lowercase only commit/PR as this affects automated release notes. `[build: Add test and build scripts]` should change to lower case only.

**Comment by sergio-costas** [comment]
Ops... sorry. Fixed.

**Pull request closed without merging**